rustls/src/client/tls13.rs

```diff
@@ -164,10 +164,7 @@ pub(super) fn handle_server_hello(
 
     #[cfg(feature = "quic")]
     {
-        cx.common.quic.hs_secrets = Some(quic::Secrets {
-            server: server_key,
-            client: client_key,
-        });
+        cx.common.quic.hs_secrets = Some(quic::Secrets::new(server_key, client_key, suite, true));
```

Comment:
Also, regardless, it seems like we need to ensure we have a test for this.

Reply:
Fixed it, will see about adding a test.

Reply:
I definitely think we should have a test for this if at all practical since it would have found a bug before code review. It would have been easy for me to miss this in the review.

Reply:
Besides the test, I filed issue #765 regarding this.

Reply:
Looks like I removed tests that could have caught this in the first commit here. I think I can just fix those up.

Reply:
Updated the tests that checked for this and verified that they would have caught this issue.